pmlc/dialect/pxa/transforms/vectorize.cc

```diff
@@ -238,10 +353,29 @@ class VectorizeCandidate {
     if (!vectorizedOps.count(op)) {
       return;
     }
-    TypeSwitch<Operation *>(op)
-        .Case<PxaLoadOp>([&](auto op) { vectorizeLoadOp(op); })
-        .Case<PxaReduceOp>([&](auto op) { vectorizeReduceOp(op); })
-        .Default([&](Operation *op) { vectorizeScalarOp(op); });
+    if (!loopMathOp) {
+      TypeSwitch<Operation *>(op)
+          .Case<PxaLoadOp>([&](auto op) { vectorizeLoadOp(op); })
+          .Case<PxaReduceOp>([&](auto op) { vectorizeReduceOp(op); })
+          .Default([&](Operation *op) { vectorizeScalarOp(op); });
+    } else {
+      TypeSwitch<Operation *>(op)
+          .Case<PxaLoadOp>([&](auto op) { vectorizeLoadOp(op); })
+          .Case<PxaReduceOp>([&](auto op) { vectorizeReduceOp(op); })
+          .Case<math::CosOp>(
+              [&](auto op) { createLoopForMathUnaryOp<math::CosOp>(op); })
+          .Case<math::ExpOp>(
+              [&](auto op) { createLoopForMathUnaryOp<math::ExpOp>(op); })
+          .Case<math::LogOp>(
+              [&](auto op) { createLoopForMathUnaryOp<math::LogOp>(op); })
+          .Case<math::SinOp>(
+              [&](auto op) { createLoopForMathUnaryOp<math::SinOp>(op); })
+          .Case<math::SqrtOp>(
+              [&](auto op) { createLoopForMathUnaryOp<math::SqrtOp>(op); })
+          .Case<math::TanhOp>(
+              [&](auto op) { createLoopForMathUnaryOp<math::TanhOp>(op); })
+          .Default([&](Operation *op) { vectorizeScalarOp(op); });
+    }
```

Comment:
Currently extra math ops, only keep Math Ops that are lowered in TileToPXA pass.

Maybe we can discuss should we keep all ops in math dialect.